nucliadb_models/src/nucliadb_models/graph/requests.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+from enum import Enum
+from typing import Annotated, Literal, Optional, Union
+
+from pydantic import BaseModel, Discriminator, Field, Tag, model_validator
+from typing_extensions import Self
+
+from nucliadb_models.filters import And, Not, Or, filter_discriminator
+from nucliadb_models.metadata import RelationNodeType
+
+## Models for graph nodes and relations
+
+
+class NodeMatchKind(str, Enum):
+    EXACT = "exact"
+    FUZZY = "fuzzy"
+
+
+class GraphNode(BaseModel):
+    value: Optional[str] = None
+    match: NodeMatchKind = NodeMatchKind.EXACT
+    type: Optional[RelationNodeType] = RelationNodeType.ENTITY
```

Comment:
if you set a default value that is not none, you can remove the Optional and assume will always be set

Reply:
I wanted to provide the usual default but allow users to specify `type=None`, i.e., any `type`. I can remove the default if that's confusing